core/src/consensus/stake/actions.rs

```diff
@@ -30,6 +32,11 @@ pub enum Action {
         address: Address,
         quantity: u64,
     },
+    ChangeParams {
+        metadata_seq: u64,
+        params: CommonParams,
```

Comment:
I think it contradicts the [spec](https://github.com/sgkim126/codechain/blob/master/spec/Staking.md#new_parameters). Spec says that the transaction contains fields that are going to be changed, however, this implementation seems to have whole `CommonParams` like a snapshot.

Reply:
Oh, I'd forgot to update the spec.
It can be dangerous to specify the field that the transaction wants to change. See the above scenario.
Assuming that the parameter X and Y contradicts in certain value and there are 2N+1 stakeholders and the parameters. If one malicious stakeholder collects the signature to change X of N stakeholders and the signature to change Y of others, only one malicious stakeholder can attack the chain.
So I changed ChangeParams have whole parameters but I forgot to update the spec. I'll update it.

Reply:
https://github.com/CodeChain-io/codechain/pull/1560 I uploaded PR.